lib/ash/resource.ex

```diff
@@ -242,11 +241,10 @@ defmodule Ash.Resource do
               }
 
               infos =
-                for %{field: field} = info <- record.__struct__.__info__(:struct),
-                    field in @show_fields,
+                for field <- @show_fields,
                     not (field == :aggregates && record.aggregates == %{}),
                     not (field == :calculations && record.calculations == %{}),
-                    do: info
+                    do: %{field: field}
```

Comment:
Couple of issues: 

1. we'll need to add `:aggregates` and `:calculations` if they aren't empty (those were in the struct fields but not `@show_fields`
2. I'm not sure what other fields are inside of the `info` that we get  back from `__info__(:struct)`, so we need to go and find that struct field's `info` and return that here.

Reply:
Thanks for the quick review :)

1. It looks like aggregates and calculations and added to `@show_fields` on line 215 (`|> Enum.concat([:__meta__, :calculations, :aggregates])`)
2. On elixir 1.18, `__info__(:struct)` is a list of `%{field: ..., default: ...}`, and only `field` is used in the `Inspect` module. But yeah, this is a bit of internal hackings. I was trying to avoid having to create a map from the `__info__(:struct)` to avoid being quadratic, but I can do that

Reply:
Oh, sorry, I missed your message. I like the idea of inlining the logic better than what I have done, it removes the need for two variants, will be overall simpler. I'll do that instead.